projects/eval-ade9153a/src/common/common_data.c

```diff
@@ -0,0 +1,407 @@
+	.number = GPIO_LED_PIN,
+	.pull = NO_OS_PULL_UP,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+// ss pin
+struct no_os_gpio_init_param gpio_ss_ip = {
+	.port = GPIO_SS_PORT,
+	.number = GPIO_SS_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+// sck pin
+struct no_os_gpio_init_param gpio_sck_ip = {
+	.port = GPIO_SCK_PORT,
+	.number = GPIO_SCK_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+// reset pin
+struct no_os_gpio_init_param gpio_reset_ip = {
+	.port = GPIO_RESET_PORT,
+	.number = GPIO_RESET_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+// rdy pin
+struct no_os_gpio_init_param ade9153a_gpio_rdy_ip = {
+	.port = GPIO_RDY_PORT,
+	.number = GPIO_RDY_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+struct no_os_spi_init_param ade9153a_spi_ip = {
+	.device_id = SPI_DEVICE_ID,
+	.max_speed_hz = SPI_BAUDRATE,
+	.bit_order = NO_OS_SPI_BIT_ORDER_MSB_FIRST,
+	.mode = NO_OS_SPI_MODE_0,
+	.platform_ops = &max_spi_ops,
+	.chip_select = SPI_CS,
+	.extra = &ade9153a_spi_extra_ip,
+};
+
+struct no_os_irq_init_param ade9153a_gpio_irq_ip = {
+	.platform_ops = GPIO_IRQ_OPS,
+	.irq_ctrl_id = GPIO_CTRL_IRQ_ID,
+	.extra = GPIO_IRQ_EXTRA,
+};
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Toggle Led
+ * @param gpio_led_desc - led descriptor
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int interface_toggle_led(struct no_os_gpio_desc *gpio_led_desc)
+{
+	uint8_t val;
+	int ret;
+
+	ret = no_os_gpio_get_value(gpio_led_desc, &val);
+	if (ret)
+		return ret;
+
+	if (val == NO_OS_GPIO_LOW)
+		val = NO_OS_GPIO_HIGH;
+	else
+		val = NO_OS_GPIO_LOW;
+
+	return no_os_gpio_set_value(gpio_led_desc, val);
+}
+
+/**
+ * @brief Read measurements
+* @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int read_measurements(struct ade9153a_dev *dev)
+{
+	int ret;
+	struct ade9153a_temperature_value temp_val;
+	struct ade9153a_energy_values energy_vals;
+	struct ade9153a_power_values power_vals;
+	struct ade9153a_rms_values rms_vals;
+	struct ade9153a_half_rms_values hrms_vals;
+	struct ade9153a_pq_values pq_vals;
+	float temperature_val;
+	float active_energy, reactive_energy, apparent_energy;
+	float active_power, reactive_power, apparent_power;
+	float current_rms_val, voltage_rms_val;
+	float current_h_rms_val, voltage_h_rms_val;
+	float constant, power_factor_val, frequency_val, angle_ai_av_val;
+	// status of freq select
+	uint8_t status;
+	// skip temp if timeout
+	uint8_t skip_temp;
+
+
+	ret = ade9153a_temp_val(dev, &temp_val);
+	if (ret)
+		if (ret == -ENODATA) {
```

Comment:
if the `if` statements has brackets then the `else` should have too (even if it is only one line)